text/0000-cache-generated-assembly.md

```diff
@@ -0,0 +1,84 @@
+- Start Date: 2018-10-31
+- RFC PR: (leave this empty)
+- Cake Issue: (leave this empty)
+- Affected Areas: Cake, Cake.Core, Cake.Scripting
+
+## Summary
+[summary]: #summary
+
+Include in Cake the ability to cache the generated assembly, so that this
+can be used when the underlying script hasn't changed.
+
+## Motivation
+[motivation]: #motivation
+
+While for most people, the speed in which Cake executes is perfectly fine,
+however, there have been those that suggest that the compilation time is a
+burden that they don't want to spend.  Especially in situations where the build
+script itself isn't changing, but lots of builds are happening in succession.
+
+Caching the generated assembly so that it can be used over and over again, without
+the need to compile each time, will save some workloads loads of time.
+
+## Detailed design
+[detailed-design]: #detailed-design
+
+### Changes to configuration
+
+A new entry will be added into the cake.config file to control the location of
+the cached assemblies.  This should default to a `cache` folder within the Cake
+Tools folder.  The end user can then control this location in the normal way.
+
+### Changes to Cake Arguments
+
+Two new arguments will be added to the Cake Client.
+
+* Cache - If set true the compiled script will be cached
```

Comment:
I think this could (should) be true by default. Maybe there can be an additional check to check modification time of files against the build time of the cached assembly to determine whether a new build is required.

Reply:
They wanted this to be opt-in so that is why it was defaulted to false.  There is an additional step to determine if the cached assembly is valid.  A hash is taken of the final script and that is used to determine if the cache is valid or not.